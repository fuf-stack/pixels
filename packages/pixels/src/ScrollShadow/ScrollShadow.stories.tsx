import type { Meta, StoryObj } from '@storybook/react';

import ScrollShadow from './ScrollShadow';

const meta: Meta<typeof ScrollShadow> = {
  title: 'pixels/ScrollShadow',
  component: ScrollShadow,
  parameters: {
    layout: 'centered',
  },
  argTypes: {
    orientation: {
      control: 'select',
      options: ['horizontal', 'vertical'],
    },
    size: {
      control: { type: 'range', min: 10, max: 100, step: 5 },
    },
    hideScrollBar: {
      control: 'boolean',
    },
  },
};

export default meta;
type Story = StoryObj<typeof ScrollShadow>;

const funnyVerticalItems = [
  '🍕 Pizza delivery guy who forgot the pizza',
  '🐢 Turtle racing champion (retired)',
  '🎵 Elevator music that makes you question life choices',
  '🧦 Sock that went missing in the dryer (found!)',
  '🚀 Rocket ship powered by coffee',
  '🐧 Penguin wearing a tuxedo to a beach party',
  '🎪 Circus elephant who learned to juggle',
  '🌮 Taco Tuesday on a Wednesday (scandal!)',
  '🎭 Drama llama with trust issues',
  '🎸 Guitar that only plays polka music',
  '🦖 Dinosaur who survived by being really good at hide and seek',
  "🍪 Cookie monster's diet coach",
  '🎯 Dart board that never gets bullseyes',
  '🎨 Paint that watched too much reality TV',
  '🧸 Teddy bear who moonlights as a bouncer',
  '🎮 Gaming controller with commitment issues',
  '🌈 Rainbow that ran out of colors',
  '🔮 Crystal ball showing only cat videos',
  "🎪 Clown who's afraid of children",
  "🐙 Octopus who can't multitask",
];

const VerticalContent = () => (
  <>
    {/* eslint-disable-next-line prefer-spread */}
    {Array.apply(null, Array(50)).map((_, i) => {
      const item = funnyVerticalItems[i % funnyVerticalItems.length];
      return (
        <div
          // eslint-disable-next-line react/no-array-index-key
          key={i}
          className="border-b border-default-200 p-2 hover:bg-default-50"
        >
          {item}
        </div>
      );
    })}
  </>
);

const funnyHorizontalCards = [
  '🎭 Drama Queen',
  '🦄 Unicorn CEO',
  '🐸 Frog Prince',
  '🎪 Circus Manager',
  '🧙‍♂️ Wizard IT',
  '🎨 Artist Barista',
  '🚀 Space Janitor',
  '🦖 Dino Lawyer',
  '🎸 Rock Star Chef',
  '🧸 Teddy Bear Therapist',
  '🎯 Dart Champion',
  '🎮 Gaming Grandma',
  '🌮 Taco Philosopher',
  '🎪 Clown Doctor',
  '🔮 Fortune Teller',
  '🐙 Octopus Accountant',
  '🎵 Musical Dentist',
  '🌈 Rainbow Engineer',
  '🍪 Cookie Architect',
  '🎭 Method Actor',
];

const HorizontalContent = () => (
  <div className="flex space-x-4">
    {/* eslint-disable-next-line prefer-spread */}
    {Array.apply(null, Array(20)).map((_, i) => {
      const card = funnyHorizontalCards[i % funnyHorizontalCards.length];
      return (
        <div
          // eslint-disable-next-line react/no-array-index-key
          key={i}
          className="h-20 w-32 flex-none cursor-pointer rounded bg-info-100 p-2 text-sm transition-colors hover:bg-info-200"
        >
          <div className="text-center font-medium">{card}</div>
        </div>
      );
    })}
  </div>
);

export const Vertical: Story = {
  args: {
    className: 'w-[300px] h-[400px]',
    children: <VerticalContent />,
  },
};

export const Horizontal: Story = {
  args: {
    className: 'w-[400px] h-[120px]',
    orientation: 'horizontal',
    children: <HorizontalContent />,
  },
};

export const HideScrollBar: Story = {
  args: {
    className: 'w-[300px] h-[300px]',
    hideScrollBar: true,
    children: <VerticalContent />,
  },
};

export const LargeShadow: Story = {
  args: {
    className: 'w-[300px] h-[400px]',
    size: 120,
    children: <VerticalContent />,
  },
};

export const SmallShadow: Story = {
  args: {
    className: 'w-[300px] h-[400px]',
    size: 15,
    children: <VerticalContent />,
  },
};
