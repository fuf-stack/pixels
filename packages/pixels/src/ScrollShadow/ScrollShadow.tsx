import type { ReactNode } from 'react';

import { ScrollShadow } from '@nextui-org/scroll-shadow';

import { cn } from '@fuf-stack/pixel-utils';

export interface ScrollShadowProps {
  /** child components */
  children?: ReactNode;
  /** CSS class name */
  className?: string | string[];
  /** content of the scroll box */
  testId?: string;
}

/**
 * Scroll Shadow component based on [NextUI Scroll Shadow](https://nextui.org/docs/components/scroll-shadow)
 */
export default ({
  children = null,
  className = undefined,
  testId = undefined,
}: ScrollShadowProps) => (
  <ScrollShadow className={cn(className)} data-testId={testId}>
    {children}
  </ScrollShadow>
);
