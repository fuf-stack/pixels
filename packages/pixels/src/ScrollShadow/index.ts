import ScrollShadow from './ScrollShadow';

export type { ScrollShadowProps } from './ScrollShadow';

export { ScrollShadow };

export default ScrollShadow;
