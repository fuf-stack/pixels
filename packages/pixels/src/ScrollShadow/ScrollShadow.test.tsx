import { describe, expect, test } from 'vitest';

import storySnapshots from '@repo/storybook-config/story-snapshots';
import { render } from '@testing-library/react';

import ScrollShadow from './ScrollShadow';
// import { render } from '@testing-library/react';

import * as stories from './ScrollShadow.stories';

describe('Story Snapshots', () => {
  storySnapshots(stories);
});

describe('Coverage', () => {
  test('renders ScrollShadow component without children and className', () => {
    const { container } = render(<ScrollShadow />);
    expect(container.firstChild).toMatchSnapshot();
  });
});
