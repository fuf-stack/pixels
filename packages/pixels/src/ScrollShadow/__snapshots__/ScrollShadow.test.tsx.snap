// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`Coverage > renders ScrollShadow component without children and className 1`] = `
<div
  class="overflow-y-auto data-[top-scroll=true]:[mask-image:linear-gradient(0deg,#000_calc(100%_-_var(--scroll-shadow-size)),transparent)] data-[bottom-scroll=true]:[mask-image:linear-gradient(180deg,#000_calc(100%_-_var(--scroll-shadow-size)),transparent)] data-[top-bottom-scroll=true]:[mask-image:linear-gradient(#000,#000,transparent_0,#000_var(--scroll-shadow-size),#000_calc(100%_-_var(--scroll-shadow-size)),transparent)]"
  data-orientation="vertical"
  style="--scroll-shadow-size: 40px;"
/>
`;

exports[`Story Snapshots > Default 1`] = `
<div>
  <div
    class="overflow-y-auto data-[top-scroll=true]:[mask-image:linear-gradient(0deg,#000_calc(100%_-_var(--scroll-shadow-size)),transparent)] data-[bottom-scroll=true]:[mask-image:linear-gradient(180deg,#000_calc(100%_-_var(--scroll-shadow-size)),transparent)] data-[top-bottom-scroll=true]:[mask-image:linear-gradient(#000,#000,transparent_0,#000_var(--scroll-shadow-size),#000_calc(100%_-_var(--scroll-shadow-size)),transparent)] w-[300px] h-[400px]"
    data-orientation="vertical"
    style="--scroll-shadow-size: 40px;"
  >
    <div>
      too much content
    </div>
    <div>
      too much content
    </div>
    <div>
      too much content
    </div>
    <div>
      too much content
    </div>
    <div>
      too much content
    </div>
    <div>
      too much content
    </div>
    <div>
      too much content
    </div>
    <div>
      too much content
    </div>
    <div>
      too much content
    </div>
    <div>
      too much content
    </div>
    <div>
      too much content
    </div>
    <div>
      too much content
    </div>
    <div>
      too much content
    </div>
    <div>
      too much content
    </div>
    <div>
      too much content
    </div>
    <div>
      too much content
    </div>
    <div>
      too much content
    </div>
    <div>
      too much content
    </div>
    <div>
      too much content
    </div>
    <div>
      too much content
    </div>
    <div>
      too much content
    </div>
    <div>
      too much content
    </div>
    <div>
      too much content
    </div>
    <div>
      too much content
    </div>
    <div>
      too much content
    </div>
    <div>
      too much content
    </div>
    <div>
      too much content
    </div>
    <div>
      too much content
    </div>
    <div>
      too much content
    </div>
    <div>
      too much content
    </div>
    <div>
      too much content
    </div>
    <div>
      too much content
    </div>
    <div>
      too much content
    </div>
    <div>
      too much content
    </div>
    <div>
      too much content
    </div>
    <div>
      too much content
    </div>
    <div>
      too much content
    </div>
    <div>
      too much content
    </div>
    <div>
      too much content
    </div>
    <div>
      too much content
    </div>
    <div>
      too much content
    </div>
    <div>
      too much content
    </div>
    <div>
      too much content
    </div>
    <div>
      too much content
    </div>
    <div>
      too much content
    </div>
    <div>
      too much content
    </div>
    <div>
      too much content
    </div>
    <div>
      too much content
    </div>
    <div>
      too much content
    </div>
    <div>
      too much content
    </div>
    <div>
      too much content
    </div>
    <div>
      too much content
    </div>
    <div>
      too much content
    </div>
    <div>
      too much content
    </div>
    <div>
      too much content
    </div>
    <div>
      too much content
    </div>
    <div>
      too much content
    </div>
    <div>
      too much content
    </div>
    <div>
      too much content
    </div>
    <div>
      too much content
    </div>
    <div>
      too much content
    </div>
    <div>
      too much content
    </div>
    <div>
      too much content
    </div>
    <div>
      too much content
    </div>
    <div>
      too much content
    </div>
    <div>
      too much content
    </div>
    <div>
      too much content
    </div>
    <div>
      too much content
    </div>
    <div>
      too much content
    </div>
    <div>
      too much content
    </div>
    <div>
      too much content
    </div>
    <div>
      too much content
    </div>
    <div>
      too much content
    </div>
    <div>
      too much content
    </div>
    <div>
      too much content
    </div>
    <div>
      too much content
    </div>
    <div>
      too much content
    </div>
    <div>
      too much content
    </div>
    <div>
      too much content
    </div>
    <div>
      too much content
    </div>
    <div>
      too much content
    </div>
    <div>
      too much content
    </div>
    <div>
      too much content
    </div>
    <div>
      too much content
    </div>
    <div>
      too much content
    </div>
    <div>
      too much content
    </div>
    <div>
      too much content
    </div>
    <div>
      too much content
    </div>
    <div>
      too much content
    </div>
    <div>
      too much content
    </div>
    <div>
      too much content
    </div>
    <div>
      too much content
    </div>
    <div>
      too much content
    </div>
    <div>
      too much content
    </div>
    <div>
      too much content
    </div>
    <div>
      too much content
    </div>
    <div>
      too much content
    </div>
    <div>
      too much content
    </div>
    <div>
      too much content
    </div>
    <div>
      too much content
    </div>
  </div>
</div>
`;
